osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java

```diff
@@ -831,32 +831,33 @@ public static String maskSecretKey(String logStatement) {
      * Converts S3 Bucket to OSIS bucket meta
      *
      * @param bucket S3 Bucket instance
+     * @param canonicalId canonicalId
      * @return the OSIS bucket meta
      */
-    public static OsisBucketMeta toOsisBucketMeta(Bucket bucket, String userId) {
+    public static OsisBucketMeta toOsisBucketMeta(Bucket bucket, String canonicalId) {
 
         return new OsisBucketMeta()
                 .name(bucket.getName())
                 .creationDate(bucket.getCreationDate().toInstant().toString())
-                .userId(userId);
+                .userId(canonicalId);
```

Comment:
What's the backward compatibility implication, if what was a userId is now the canonical id? What was used as a user id before?

Reply:
This was never implemented properly by VMware OSE side. Previously we were sending account ID as owner/user ID and they were expecting buckets to be owned by IAM users like in ceph model. Hence this API never worked with VCD and as a result end users were never able to synchronize buckets in VCD. We collaborated with VMware engg last year for 4+ months helping them get true IAM support for OSE.  Per IAM spec they now support buckets being owned by accounts ie canonical ID. Now this API will be honored in OSE2.2.0.1 for true IAM style ownership.

Reply:
As this API was never used by OSE I do not see any backward compatibility support needed for this. OSE does rebuild database entries on trying bucket synchronization on OSE(tested by us and VMware engg). It maintains its own entries for ownership in a Postgres server.

Reply:
This is the new property `iam: true` in getInformation API, which tells OSE (support from v2.2.0.1+) to flip the switch from ceph to supporting IAM standard.